File: apps/backend/source/types/chains.ts
import { Chain as ViemChain } from 'viem/chains'

export const enum CHAIN_ID {
  ETHEREUM = 1,
  SEPOLIA = 11155111,
  OPTIMISM = 10,
  OPTIMISM_SEPOLIA = 11155420,
  BASE = 8453,
  BASE_SEPOLIA = 84532,
  ZORA = 7777777,
  ZORA_SEPOLIA = 999999999,
  FOUNDRY = 31337
}

export interface Chain extends ViemChain {
  id: CHAIN_ID
  slug: string
  icon: string
}

File: apps/mobileapp/ios/Data/Chains.swift
//
//  Chains.swift
//  mobileapp
//
//  Created by NG on 22.07.2024.
//

import Foundation

enum ChainID: Int {
  case ethereum = 1
  case optimism = 10
  case base = 8453
  case zora = 7777777
  
  var stringValue: String {
    switch self {
    case .ethereum: return "ethereum"
    case .optimism: return "optimism"
    case .base: return "base"
    case .zora: return "zora"
    }
  }
}
